aiokafka/consumer.py

```diff
@@ -1,375 +1,623 @@
+        be none if the assignment hasn't happened yet or if the partitions are
+        in the process of being reassigned).
 
-class AIOConsumer(object):
+        Returns:
+            set: {TopicPartition, ...}
+        """
+        return self._subscription.assigned_partitions()
 
-    AUTO_COMMIT_MSG_COUNT = 100
-    AUTO_COMMIT_INTERVAL = 5000
-    FETCH_DEFAULT_BLOCK_TIMEOUT = 1
-    FETCH_BUFFER_SIZE_BYTES = 4096
-    MAX_FETCH_BUFFER_SIZE_BYTES = FETCH_BUFFER_SIZE_BYTES * 8
+    @asyncio.coroutine
+    def stop(self):
+        """Close the consumer, waiting indefinitely for any needed cleanup."""
+        if self._closed:
+            return
+        log.debug("Closing the KafkaConsumer.")
+        self._closed = True
+        if self._coordinator:
+            yield from self._coordinator.close()
+        if self._fetcher:
+            yield from self._fetcher.close()
+        self._client.close()
+        try:
+            self._key_deserializer.close()
+        except AttributeError:
+            pass
+        try:
+            self._value_deserializer.close()
+        except AttributeError:
+            pass
+        log.debug("The KafkaConsumer has closed.")
 
-    def __init__(self, client, group,
-                 topic,
-                 partitions=None,
-                 auto_commit=True,
-                 auto_commit_every_n=AUTO_COMMIT_MSG_COUNT,
-                 auto_commit_every_t=AUTO_COMMIT_INTERVAL):
+    @asyncio.coroutine
+    def commit(self, offsets=None):
+        """Commit offsets to kafka, blocking until success or error
 
-        self._client = client
-        self._loop = self._client.loop
+        This commits offsets only to Kafka.
+        The offsets committed using this API will be used on the first fetch
+        after every rebalance and also on startup. As such, if you needto store
+        offsets in anything other than Kafka, this API should not be used.
 
-        self._topic = topic
-        self._group = group
-        self._offsets = {}
+        Blocks until either the commit succeeds or an unrecoverable error is
+        encountered (in which case it is thrown to the caller).
 
-        # Variables for handling offset commits
-        self._commit_lock = asyncio.Lock(loop=self._loop)
-        self._commit_timer = None
-        self._count_since_commit = 0
-        self._auto_commit = auto_commit
-        self._auto_commit_every_n = auto_commit_every_n
-        self._auto_commit_every_t = auto_commit_every_t
+        Currently only supports kafka-topic offset storage (not zookeeper)
 
-        self._partitions = partitions
-        self._commit_task = None
+        Arguments:
+            offsets (dict, optional): {TopicPartition: OffsetAndMetadata} dict
+                to commit with the configured group_id. Defaults to current
+                consumed offsets for all subscribed partitions.
+        """
+        assert self._group_id is not None, 'Requires group_id'
```

Comment:
How do you think, should we allow to save offsets, that are not part of our assigned partitions?